responsibleai/tests/test_rai_insights.py

```diff
@@ -167,6 +167,104 @@ def test_modelanalysis_boston(self, manager_type):
+
+        models = create_models_classification(X_train, y_train)
+        X_train[LABELS] = y_train
+        X_test[LABELS] = y_test
+
+        for model in models:
+            rai_insights = RAIInsights(
+                model, X_train, X_test,
+                LABELS,
+                categorical_features=None,
+                task_type=ModelTask.CLASSIFICATION)
+
+            with TemporaryDirectory() as tmpdir:
+                save_1 = Path(tmpdir) / "first_save"
+                save_2 = Path(tmpdir) / "second_save"
+
+                # Save it
+                rai_insights.save(save_1)
+
+                # Load
+                rai_2 = RAIInsights.load(save_1)
+
+                # Validate, but this isn't the main check
+                validate_rai_insights(
+                    rai_2, X_train, X_test,
+                    LABELS, ModelTask.CLASSIFICATION, None)
+
+                # Save again (this is where Issue #1046 manifested)
+                rai_2.save(save_2)
+
+    @pytest.mark.parametrize('manager_type', [ManagerNames.CAUSAL,
+                                              ManagerNames.ERROR_ANALYSIS,
+                                              ManagerNames.EXPLAINER,
+                                              ManagerNames.COUNTERFACTUAL])
+    def test_rai_insights_save_load_add_save(self, manager_type):
+        data_train, data_test, y_train, y_test, categorical_features, \
+            continuous_features, target_name, classes = \
+            create_adult_income_dataset()
+        X_train = data_train.drop([target_name], axis=1)
+
+        model = create_complex_classification_pipeline(
+            X_train, y_train, continuous_features, categorical_features)
+
+        # Cut down size for counterfactuals, in the interests of speed
+        if manager_type == ManagerNames.COUNTERFACTUAL:
+            data_test = data_test[0:1]
+
+        rai_insights = RAIInsights(
+            model, data_train, data_test,
+            target_name,
+            categorical_features=categorical_features,
+            task_type=ModelTask.CLASSIFICATION)
+
+        with TemporaryDirectory() as tmpdir:
+            save_1 = Path(tmpdir) / "first_save"
+            save_2 = Path(tmpdir) / "second_save"
+
+            # Save it
+            rai_insights.save(save_1)
+
+            # Load
+            rai_2 = RAIInsights.load(save_1)
+
+            # Call a single manager
+            if manager_type == ManagerNames.CAUSAL:
+                rai_2.causal.add(
+                    treatment_features=['age', 'hours_per_week']
+                )
+            elif manager_type == ManagerNames.COUNTERFACTUAL:
+                rai_2.counterfactual.add(
+                    total_CFs=10,
+                    desired_class='opposite',
+                    feature_importance=False
+                )
+            elif manager_type == ManagerNames.ERROR_ANALYSIS:
+                rai_2.error_analysis.add(
```

Comment:
looks like the parameters are missing, or the extra blank whitespace should be removed?

Reply:
Good point. Fixed